# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/angle/checkout/src/common/third_party/base/anglebase/numerics/safe_math.h
# Commit: 13282d7a47a5
# Full Hash: 13282d7a47a57a73be8d060efd9645b4d642f03e
# Author: Miko Mynttinen <mikokm@gmail.com>
# Date: 2019-09-08 09:46:52
# Regressor Bug: 1578576
# File Overlap Count: 4
# Description:
#   Bug 1578576 - Part 2: Update to ANGLE 3865 r=jgilbert
#   
#   Depends on D44561
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D44579
# ==============================================================================

diff -r 3064469c073d -r 13282d7a47a5 gfx/angle/checkout/src/common/third_party/base/anglebase/numerics/safe_math.h
--- a/gfx/angle/checkout/src/common/third_party/base/anglebase/numerics/safe_math.h	Sat Sep 07 13:36:46 2019 +0000
+++ b/gfx/angle/checkout/src/common/third_party/base/anglebase/numerics/safe_math.h	Sat Sep 07 13:38:36 2019 +0000
@@ -89,6 +89,14 @@
     // IsValid() is the public API to test if a CheckedNumeric is currently valid.
     bool IsValid() const { return validity() == RANGE_VALID; }
 
+    // AssignIfValid(Dst) - Assigns the underlying value if it is currently valid and is within the
+    // range supported by the destination type. Returns true if successful and false otherwise.
+    template <typename Dst>
+    constexpr bool AssignIfValid(Dst *result) const
+    {
+        return IsValid() ? ((*result = static_cast<Dst>(state_.value())), true) : false;
+    }
+
     // ValueOrDie() The primary accessor for the underlying value. If the current
     // state is not valid it will CHECK and crash.
     T ValueOrDie() const